### ISSUE TITLE
miniaudio support?

### ISSUE THREAD

**Issue by amerkoleci** [NazaraEngine/NazaraEngine#433]
Hi,
Excellent stuff, are there any plans to replace OpenAL (which doesn't allow static linking) with [miniaudio](https://github.com/mackron/miniaudio) ?

Thanks,
Amer

**Comment by SirLynix** [comment]
Hello,

Thanks a lot!
I didn't have plans to replace OpenAL with miniaudio because I had no issue with it (despite its LGPL license), but I'm open to it (maybe first by supporting both implementations in the Audio module?).

I don't know miniaudio well, could you tell me what are its advantages/drawbacks relative to OpenAL Soft?

Thank you

**Comment by amerkoleci** [comment]
Hi,
First of all the license, second miniaudio comes with wav, flac, mp3 and ogg support (if using the high level engine), it has multiple listener support and supports 3D audio as well, it supports also other goodies (opus encoding) but also wave effects and reverb etc, not sure what the drawbacks over OpenAL soft are, for me it's always been the LGPL license.

Cheers,
Amer

**Comment by SirLynix** [comment]
I see.

I already have wav/flac/mp3/ogg loading (thanks to dr_wav/dr_mp3 by the same author as miniaudio), I've taken a small look at minimp3 and it seems quite nice indeed, the node graph, exposed job system and the fact you don't need to handle additional features through extensions look quite nice too.
It seems the only drawbacks (relative to OpenAL) are the missing HRTF support and some advanced filters/effects, but I don't need them (nor expose them) at the moment.

I'm thinking about giving it a try, however it means rewriting the whole Audio module, not a big deal but it'll take a bit of time.

Thank you for the feature request.

**Comment by amerkoleci** [comment]
I saw this has been done,

Thanks!

**Comment by SirLynix** [comment]
Yes indeed, it took a bit more time than I thought because I had to wrap my mind around miniaudio integration and had to work on other stuff as well.

miniaudio is really a nice library, I hope I'm using it correctly. Thanks again for the suggestion.

**Comment by amerkoleci** [comment]
> Yes indeed, it took a bit more time than I thought because I had to wrap my mind around miniaudio integration and had to work on other stuff as well.
> 
> miniaudio is really a nice library, I hope I'm using it correctly. Thanks again for the suggestion.

You did well, didn't took look closer, does the audio integration support streaming? like big audio files as background audio? microphone recording?

Thanks again

**Comment by SirLynix** [comment]
Yes streaming is supported but the stream(/file) reading happens in the audio thread (when we run out of samples as there's a cache) which is problematic, I'd like to improve that in the future.

No microphone recording yet because I wanted to merge this quickly but I saw it's quite easy to do with miniaudio
